content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,228 +11,191 @@ weight: 10
+### Kubernetes core volume types {#volume-types-core}
 
-### azureDisk {#azuredisk}
+Kubernetes supports several types of volumes. Kubernetes core volume types are specialized volume types (mostly ephemeral) and are not used for general purpose persistent storage use cases. 
 
-The `azureDisk` volume type mounts a Microsoft Azure [Data Disk](https://docs.microsoft.com/en-us/azure/aks/csi-storage-drivers) into a pod.
+Core Volume Types:
 
-For more details, see the [`azureDisk` volume plugin](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/staging/volumes/azure_disk/README.md).
+   * [configMap](#configmap)
+   * [downwardAPI](#downwardapi)
+   * [emptyDir](#emptydir)
+   * [gitRepo (deprecated)](#gitrepo)
+   * [glusterfs](#glusterfs)
+   * [hostPath](#hostpath)
+   * [local](#local)
+   * [projected](#projected)
+   * [secret](#secret)
 
-#### azureDisk CSI migration
 
-{{< feature-state for_k8s_version="v1.19" state="beta" >}}
+### Storage plugins
 
-The `CSIMigration` feature for `azureDisk`, when enabled, redirects all plugin operations
-from the existing in-tree plugin to the `disk.csi.azure.com` Container
-Storage Interface (CSI) Driver. In order to use this feature, the [Azure Disk CSI
-Driver](https://github.com/kubernetes-sigs/azuredisk-csi-driver)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationAzureDisk`
-features must be enabled.
+A [persistentVolumeClaim](#persistentvolumeclaim) is a declarative way to ask for storage, without preference for the underlying implementation. 
 
-### azureFile {#azurefile}
+This is powered by storage plugins. Kubernetes implements the {{< glossary_tooltip text="Container Storage Interface" term_id="csi" >}} (CSI),
+which lets developers implement storage plugins without needing to make changes to the code of
+Kubernetes itself. Visit
+[Kubernetes Container Storage Interface Documentation])(https://kubernetes-csi.github.io/docs/)
+to learn more about CSI. 
 
-The `azureFile` volume type mounts a Microsoft Azure File volume (SMB 2.1 and 3.0)
-into a pod.
+In contrast to the CSI, some storage plugins are "in-tree" plugins. That is, storage technology specific code inserted into the code of Kubernetes itself, without using the CSI. The C  
 
-For more details, see the [`azureFile` volume plugin](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/staging/volumes/azure_file/README.md).
+[FlexVolume](#flexvolume) is yet another mechanism for storage plugins. Unlike CSI, FlexVolume is
+more closely tied to Kubernetes and its storage implementation. FlexVolume was developed before CSI.
 
-#### azureFile CSI migration
+Storage plugin interfaces:
 
-{{< feature-state for_k8s_version="v1.21" state="beta" >}}
+   * [CSI](#csi)
+   * [FlexVolume](#flexvolume)
+   * [persistentVolumeClaim](#persistentvolumeclaim)
```

Comment:
We could split “in tree” into core types like the downward API vs legacy vendor plugins that are in-tree because of their age, but would now not be accepted (you'd be asked to have a good long think about using CSI, right?)